File: src/transformers/IProgrammerProps.ts
import type ts from "typescript";

import type { ITypiaContext } from "./ITypiaContext";

export interface IProgrammerProps {
  context: ITypiaContext;
  modulo: ts.LeftHandSideExpression;
  type: ts.Type;
  name: string | undefined;
  init?: ts.Expression | undefined;
}

File: packages/typia/native/core/context/IProgrammerProps.go
package context

import (
  shimast "github.com/microsoft/typescript-go/shim/ast"
  shimchecker "github.com/microsoft/typescript-go/shim/checker"
)

type IProgrammerProps struct {
  Context ITypiaContext
  Modulo  *shimast.Node
  Type    *shimchecker.Type
  Name    *string
  Init    *shimast.Node
}
